vyper/codegen/expr.py

```diff
@@ -890,10 +890,13 @@ def parse_Compare(self):
             right = Expr(self.expr.right, self.context).lll_node
 
             length_mismatch = left.typ.maxlen != right.typ.maxlen
+            empty_eq_length_array = left.typ.maxlen == right.typ.maxlen and (
+                right.value == "~empty" or left.value == "~empty"
+            )
             left_over_32 = left.typ.maxlen > 32
             right_over_32 = right.typ.maxlen > 32
 
-            if length_mismatch or left_over_32 or right_over_32:
+            if length_mismatch or empty_eq_length_array or left_over_32 or right_over_32:
```

Comment:
no need to use keccak if one side is `empty`, can just check that the length of the other side is 0.